tests/python/gaia-ui-tests/gaiatest/form_controls/header.py

```diff
@@ -0,0 +1,38 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from abc import ABCMeta, abstractproperty
+from marionette_driver import By, Wait, expected
+from gaiatest.form_controls.form_control import Widget
+
+
+class GaiaHeader(Widget):
+    # This represents the gaia-header element
+
+    __metaclass__ = ABCMeta
+
+    def go_back(self):
+        Wait(self.marionette).until(expected.element_enabled(self.root_element) and
+                                    expected.element_displayed(self.root_element))
+        # TODO: remove tap with coordinates after Bug 1061698 is fixed
+        self.root_element.tap(x=20)
+        Wait(self.marionette).until(expected.element_not_displayed(self.root_element))
+
+    def go_back_and_exit(self, app=None, add_statusbar_height=True):
+        self.tap_element_from_system_app(self.root_element, add_statusbar_height=add_statusbar_height, x=20)
+        app.wait_to_not_be_displayed()
+        self.apps.switch_to_displayed_app()
+
+
+class HTMLHeader(GaiaHeader):
+    # This represents the header element
+
+    _back_button_locator = (By.CSS_SELECTOR, '.back')
+
+    def go_back(self):
+        back_button = self.root_element.find_element(*self._back_button_locator)
```

Comment:
The back button in GaiaHeader is not accessible (yet), because it is in shadow dom. I guess in practice it doesn't matter and I can create a combined function.